### ISSUE TITLE
Generalizing the component library

### ISSUE THREAD

**Issue by obra** [BathroomEpiphanies/KiCAD-Keyboard-Tutorial#1]
Poking around, I haven't found a general cherry keyswitch library for KiCAD other than this one.

I've been working on generalizing the component library you use in the tutorial, with an eye to making it available to GH/DT/everybody under a liberal license.

So far, I've been playing with:
- making the switch footprints reversible so they can be used for keyboards like the ergodox
- making variants of the switch footprints with builtin pin placement for diodes (through-hole, SMD and in-switch) - This also means that designers could build 'simpler' schematics with half the component and wire count.
- making variants of the switch footprints with 18mm keycap outlines

Does this all just sound like a bad idea?

Am I missing a better general keyswitch component library?

Should I be using some other keyboard as a starting point?

Is it worth me submitting these as a pull request to footprints or would it be better to create a standalone github repo? 

Thanks!

**Comment by kejadlen** [comment]
@technomancy What did you wind up doing for the Atreus? Did the footprints for the switches in the [Ergodox PCB](http://ergodox.org/Downloads/ErgoDox_PCB_Designed_by_pbiphany.zip) not work either? (I'm using those, so would like to know if they aren't feasible for some reason.) The Ergodox is GPL'ed, although I'm not sure if that works for you or not. Kind of a moot point now, I guess!

**Comment by technomancy** [comment]
Yeah, I ended up using the Ergodox footprint. The Atreus is GPLed, so it's a perfect fit. However, I haven't yet tested it for Alps compatibility. I hope to do so in a month or two.

**Comment by kejadlen** [comment]
@technomancy Whew, thanks - I'm glad to hear that.
